mis_builder/models/aep.py

```diff
@@ -368,17 +294,146 @@ def f(mo):
             field, mode, account_codes, domain = self._parse_match_object(mo)
             key = (domain, mode)
             account_ids_data = self._data[key]
-            v = 0.0
+            v = AccountingNone
             for account_code in account_codes:
                 account_ids = self._account_ids_by_code[account_code]
                 for account_id in account_ids:
                     debit, credit = \
-                        account_ids_data.get(account_id, (0.0, 0.0))
+                        account_ids_data.get(account_id,
+                                             (AccountingNone, AccountingNone))
                     if field == 'bal':
                         v += debit - credit
                     elif field == 'deb':
                         v += debit
                     elif field == 'crd':
                         v += credit
+            # in initial balance mode, assume 0 is None
+            # as it does not make sense to distinguish 0 from "no data"
+            if v is not AccountingNone and \
+                    mode in (self.MODE_INITIAL, self.MODE_UNALLOCATED) and \
+                    float_is_zero(v, precision_rounding=2):
+                v = AccountingNone
+            return '(' + repr(v) + ')'
+
+        return self._ACC_RE.sub(f, expr)
+
+    def replace_exprs_by_account_id(self, exprs):
+        """Replace accounting variables in a list of expression
+        by their amount, iterating by accounts involved in the expression.
+
+        yields account_id, replaced_expr
+
+        This method must be executed after do_queries().
+        """
+        def f(mo):
+            field, mode, account_codes, domain = self._parse_match_object(mo)
+            key = (domain, mode)
+            account_ids_data = self._data[key]
+            debit, credit = \
+                account_ids_data.get(account_id,
+                                     (AccountingNone, AccountingNone))
+            if field == 'bal':
+                v = debit - credit
+            elif field == 'deb':
+                v = debit
+            elif field == 'crd':
+                v = credit
+            # in initial balance mode, assume 0 is None
+            # as it does not make sense to distinguish 0 from "no data"
+            if v is not AccountingNone and \
+                    mode in (self.MODE_INITIAL, self.MODE_UNALLOCATED) and \
+                    float_is_zero(v, precision_rounding=2):
```

Comment:
@elicoidal you are right! in this area of mis_builder, we rely on accounting precision, not the precision set in the styles. So I changed to use the decimal places of the company currency, and adapted the AEP api to pass the company in the constructor instead of moving it around in each method.